planner/core/logical_plan_builder.go

```diff
@@ -6826,6 +6827,15 @@ func (b *PlanBuilder) buildWindowFunctionFrameBound(_ context.Context, spec *ast
 	if err != nil {
 		return nil, err
 	}
+	if bound.CalcFuncs[0].GetType().EvalType() != col.GetType().EvalType() {
```

Comment:
Add some tests to test different fieldType but the same evalType, for example, test the case that the order column is tinyint

Reply:
> Add some tests to test different fieldType but the same evalType, for example, test the case that the order column is tinyint

Done

`tk.MustQuery("select *, first_value(v) over (partition by p order by o range between 3 preceding and 2.9 following) as a from test.first_range;")`, for preceding, it will be tinyint vs bigint, for following it will be tinyint vs decimal.